src/main/java/emissary/util/PayloadUtil.java

```diff
@@ -26,6 +30,55 @@ public class PayloadUtil {
     private static final String LS = System.getProperty("line.separator");
     private static final Pattern validFormRegex = Pattern.compile("^[\\w-)(/]+$");
 
+    protected static Map<String, String> historyPreference = new HashMap<>();
+
+    protected static final String REDUCED_HISTORY = "REDUCED_HISTORY";
+    protected static final String NO_URL = "NO_URL";
+
+    static {
+        configure();
+    }
+
+    protected static void configure() {
+        Configurator configG = null;
+        try {
+            configG = ConfigUtil.getConfigInfo(PayloadUtil.class);
+        } catch (IOException e) {
+            logger.error("Cannot open default config file", e);
+        }
+
+        // fill lists with data from PayloadUtil.cfg for transform itinerary/history
+        if (configG != null) {
+            for (String fileType : configG.findEntries(REDUCED_HISTORY)) {
+                setFileTypeHistoryPreference(fileType, REDUCED_HISTORY);
+            }
+            for (String fileType : configG.findEntries(NO_URL)) {
+                setFileTypeHistoryPreference(fileType, NO_URL);
+            }
+        }
+    }
+
+    /**
+     * Check if fileType is already mapped to history preference. If not, set fileType -$gt; preference in map
+     *
+     * @param fileType current fileType pulled from cfg
+     * @param preference preference associated with current cfg fileType
+     */
+    protected static void setFileTypeHistoryPreference(String fileType, String preference) {
+        if (historyPreference.containsKey(fileType)) {
```

Comment:
In the future (i.e., not this PR), we could add an option to fail initialization if a fileType had more than one entry, but that's not something I want called from a static initialization block.